// @aws-sdk/client-kms/dist/es/commands/ListResourceTagsCommand.js
import { __extends } from "tslib";
import { ListResourceTagsRequest, ListResourceTagsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListResourceTagsCommand, serializeAws_json1_1ListResourceTagsCommand, } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Returns a list of all tags for the specified customer master key (CMK).</p>
 *          <p>You cannot perform this operation on a CMK in a different AWS account.</p>
 */
var ListResourceTagsCommand = /** @class */ (function (_super) {
    __extends(ListResourceTagsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListResourceTagsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ListResourceTagsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "ListResourceTagsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListResourceTagsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListResourceTagsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListResourceTagsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListResourceTagsCommand(input, context);
    };
    ListResourceTagsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListResourceTagsCommand(output, context);
    };
    return ListResourceTagsCommand;
}($Command));
export { ListResourceTagsCommand };
