// @aws-sdk/client-ec2/dist/es/commands/AllocateHostsCommand.js
import { __extends } from "tslib";
import { AllocateHostsRequest, AllocateHostsResult } from "../models/models_0";
import { deserializeAws_ec2AllocateHostsCommand, serializeAws_ec2AllocateHostsCommand } from "../protocols/Aws_ec2";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Allocates a Dedicated Host to your account. At a minimum, specify the supported instance
 *         	type or instance family, the Availability Zone in which to allocate the host, and the number of
 *         	hosts to allocate.</p>
 */
var AllocateHostsCommand = /** @class */ (function (_super) {
    __extends(AllocateHostsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function AllocateHostsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    AllocateHostsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "AllocateHostsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AllocateHostsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AllocateHostsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AllocateHostsCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2AllocateHostsCommand(input, context);
    };
    AllocateHostsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2AllocateHostsCommand(output, context);
    };
    return AllocateHostsCommand;
}($Command));
export { AllocateHostsCommand };
