// @aws-sdk/client-ec2/dist/es/commands/DescribeInstanceTypeOfferingsCommand.js
import { __extends } from "tslib";
import { DescribeInstanceTypeOfferingsRequest, DescribeInstanceTypeOfferingsResult } from "../models/models_2";
import { deserializeAws_ec2DescribeInstanceTypeOfferingsCommand, serializeAws_ec2DescribeInstanceTypeOfferingsCommand, } from "../protocols/Aws_ec2";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Returns a list of all instance types offered. The results can be filtered by location (Region or Availability
 *    Zone). If no location is specified, the instance types offered in the current Region are returned.</p>
 */
var DescribeInstanceTypeOfferingsCommand = /** @class */ (function (_super) {
    __extends(DescribeInstanceTypeOfferingsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeInstanceTypeOfferingsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeInstanceTypeOfferingsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DescribeInstanceTypeOfferingsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeInstanceTypeOfferingsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeInstanceTypeOfferingsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeInstanceTypeOfferingsCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DescribeInstanceTypeOfferingsCommand(input, context);
    };
    DescribeInstanceTypeOfferingsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DescribeInstanceTypeOfferingsCommand(output, context);
    };
    return DescribeInstanceTypeOfferingsCommand;
}($Command));
export { DescribeInstanceTypeOfferingsCommand };
