// @aws-sdk/client-ec2/dist/es/commands/DescribeStaleSecurityGroupsCommand.js
import { __extends } from "tslib";
import { DescribeStaleSecurityGroupsRequest, DescribeStaleSecurityGroupsResult } from "../models/models_3";
import { deserializeAws_ec2DescribeStaleSecurityGroupsCommand, serializeAws_ec2DescribeStaleSecurityGroupsCommand, } from "../protocols/Aws_ec2";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>[VPC only] Describes the stale security group rules for security groups in a specified VPC. Rules are stale when they reference a deleted security group in a peer VPC, or a security group in a peer VPC for which the VPC peering connection has been deleted.</p>
 */
var DescribeStaleSecurityGroupsCommand = /** @class */ (function (_super) {
    __extends(DescribeStaleSecurityGroupsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeStaleSecurityGroupsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeStaleSecurityGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DescribeStaleSecurityGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeStaleSecurityGroupsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeStaleSecurityGroupsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeStaleSecurityGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DescribeStaleSecurityGroupsCommand(input, context);
    };
    DescribeStaleSecurityGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DescribeStaleSecurityGroupsCommand(output, context);
    };
    return DescribeStaleSecurityGroupsCommand;
}($Command));
export { DescribeStaleSecurityGroupsCommand };
