// @aws-sdk/client-eks/dist/es/commands/DescribeClusterCommand.js
import { __extends } from "tslib";
import { DescribeClusterRequest, DescribeClusterResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeClusterCommand, serializeAws_restJson1DescribeClusterCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Returns descriptive information about an Amazon EKS cluster.</p>
 *         <p>The API server endpoint and certificate authority data returned by this operation are
 *             required for <code>kubelet</code> and <code>kubectl</code> to communicate with your
 *             Kubernetes API server. For more information, see <a href="https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html">Create a
 *                 kubeconfig for Amazon EKS</a>.</p>
 *         <note>
 *             <p>The API server endpoint and certificate authority data aren't available until the
 *                 cluster reaches the <code>ACTIVE</code> state.</p>
 *         </note>
 */
var DescribeClusterCommand = /** @class */ (function (_super) {
    __extends(DescribeClusterCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeClusterCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EKSClient";
        var commandName = "DescribeClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeClusterRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeClusterResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeClusterCommand(input, context);
    };
    DescribeClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeClusterCommand(output, context);
    };
    return DescribeClusterCommand;
}($Command));
export { DescribeClusterCommand };
