// @aws-sdk/client-ec2/dist/es/commands/DeregisterInstanceEventNotificationAttributesCommand.js
import { __extends } from "tslib";
import { DeregisterInstanceEventNotificationAttributesRequest, DeregisterInstanceEventNotificationAttributesResult, } from "../models/models_2";
import { deserializeAws_ec2DeregisterInstanceEventNotificationAttributesCommand, serializeAws_ec2DeregisterInstanceEventNotificationAttributesCommand, } from "../protocols/Aws_ec2";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Deregisters tag keys to prevent tags that have the specified tag keys from being included
 * 			in scheduled event notifications for resources in the Region.</p>
 */
var DeregisterInstanceEventNotificationAttributesCommand = /** @class */ (function (_super) {
    __extends(DeregisterInstanceEventNotificationAttributesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DeregisterInstanceEventNotificationAttributesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DeregisterInstanceEventNotificationAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DeregisterInstanceEventNotificationAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeregisterInstanceEventNotificationAttributesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeregisterInstanceEventNotificationAttributesResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeregisterInstanceEventNotificationAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DeregisterInstanceEventNotificationAttributesCommand(input, context);
    };
    DeregisterInstanceEventNotificationAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DeregisterInstanceEventNotificationAttributesCommand(output, context);
    };
    return DeregisterInstanceEventNotificationAttributesCommand;
}($Command));
export { DeregisterInstanceEventNotificationAttributesCommand };
