// @aws-sdk/client-ec2/dist/es/commands/DescribeRegionsCommand.js
import { __extends } from "tslib";
import { DescribeRegionsRequest, DescribeRegionsResult } from "../models/models_3";
import { deserializeAws_ec2DescribeRegionsCommand, serializeAws_ec2DescribeRegionsCommand } from "../protocols/Aws_ec2";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Describes the Regions that are enabled for your account, or all Regions.</p>
 *          <p>For a list of the Regions supported by Amazon EC2, see <a href="https://docs.aws.amazon.com/general/latest/gr/rande.html#ec2_region">
 *        Regions and Endpoints</a>.</p>
 *          <p>For information about enabling and disabling Regions for your account, see <a href="https://docs.aws.amazon.com/general/latest/gr/rande-manage.html">Managing AWS Regions</a> in the <i>AWS General Reference</i>.</p>
 */
var DescribeRegionsCommand = /** @class */ (function (_super) {
    __extends(DescribeRegionsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeRegionsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeRegionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DescribeRegionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeRegionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeRegionsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeRegionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DescribeRegionsCommand(input, context);
    };
    DescribeRegionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DescribeRegionsCommand(output, context);
    };
    return DescribeRegionsCommand;
}($Command));
export { DescribeRegionsCommand };
