// @aws-sdk/client-ec2/dist/es/commands/DescribeTransitGatewayVpcAttachmentsCommand.js
import { __extends } from "tslib";
import { DescribeTransitGatewayVpcAttachmentsRequest, DescribeTransitGatewayVpcAttachmentsResult, } from "../models/models_3";
import { deserializeAws_ec2DescribeTransitGatewayVpcAttachmentsCommand, serializeAws_ec2DescribeTransitGatewayVpcAttachmentsCommand, } from "../protocols/Aws_ec2";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Describes one or more VPC attachments. By default, all VPC attachments are described.
 *          Alternatively, you can filter the results.</p>
 */
var DescribeTransitGatewayVpcAttachmentsCommand = /** @class */ (function (_super) {
    __extends(DescribeTransitGatewayVpcAttachmentsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeTransitGatewayVpcAttachmentsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeTransitGatewayVpcAttachmentsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DescribeTransitGatewayVpcAttachmentsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeTransitGatewayVpcAttachmentsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeTransitGatewayVpcAttachmentsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeTransitGatewayVpcAttachmentsCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DescribeTransitGatewayVpcAttachmentsCommand(input, context);
    };
    DescribeTransitGatewayVpcAttachmentsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DescribeTransitGatewayVpcAttachmentsCommand(output, context);
    };
    return DescribeTransitGatewayVpcAttachmentsCommand;
}($Command));
export { DescribeTransitGatewayVpcAttachmentsCommand };
