// @aws-sdk/client-ec2/dist/es/commands/DetachInternetGatewayCommand.js
import { __extends } from "tslib";
import { DetachInternetGatewayRequest } from "../models/models_3";
import { deserializeAws_ec2DetachInternetGatewayCommand, serializeAws_ec2DetachInternetGatewayCommand, } from "../protocols/Aws_ec2";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Detaches an internet gateway from a VPC, disabling connectivity between the internet
 * 			and the VPC. The VPC must not contain any running instances with Elastic IP addresses or
 * 			public IPv4 addresses.</p>
 */
var DetachInternetGatewayCommand = /** @class */ (function (_super) {
    __extends(DetachInternetGatewayCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DetachInternetGatewayCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DetachInternetGatewayCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DetachInternetGatewayCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DetachInternetGatewayRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DetachInternetGatewayCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DetachInternetGatewayCommand(input, context);
    };
    DetachInternetGatewayCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DetachInternetGatewayCommand(output, context);
    };
    return DetachInternetGatewayCommand;
}($Command));
export { DetachInternetGatewayCommand };
