// @aws-sdk/client-ec2/dist/es/commands/ResetNetworkInterfaceAttributeCommand.js
import { __extends } from "tslib";
import { ResetNetworkInterfaceAttributeRequest } from "../models/models_4";
import { deserializeAws_ec2ResetNetworkInterfaceAttributeCommand, serializeAws_ec2ResetNetworkInterfaceAttributeCommand, } from "../protocols/Aws_ec2";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Resets a network interface attribute. You can specify only one attribute at a time.</p>
 */
var ResetNetworkInterfaceAttributeCommand = /** @class */ (function (_super) {
    __extends(ResetNetworkInterfaceAttributeCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ResetNetworkInterfaceAttributeCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ResetNetworkInterfaceAttributeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "ResetNetworkInterfaceAttributeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ResetNetworkInterfaceAttributeRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ResetNetworkInterfaceAttributeCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2ResetNetworkInterfaceAttributeCommand(input, context);
    };
    ResetNetworkInterfaceAttributeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2ResetNetworkInterfaceAttributeCommand(output, context);
    };
    return ResetNetworkInterfaceAttributeCommand;
}($Command));
export { ResetNetworkInterfaceAttributeCommand };
