// @aws-sdk/client-kms/dist/es/commands/ListKeysCommand.js
import { __extends } from "tslib";
import { ListKeysRequest, ListKeysResponse } from "../models/models_0";
import { deserializeAws_json1_1ListKeysCommand, serializeAws_json1_1ListKeysCommand } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Gets a list of all customer master keys (CMKs) in the caller's AWS account and
 *       Region.</p>
 */
var ListKeysCommand = /** @class */ (function (_super) {
    __extends(ListKeysCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListKeysCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ListKeysCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "ListKeysCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListKeysRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListKeysResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListKeysCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListKeysCommand(input, context);
    };
    ListKeysCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListKeysCommand(output, context);
    };
    return ListKeysCommand;
}($Command));
export { ListKeysCommand };
