// @aws-sdk/client-ec2/dist/cjs/commands/DeregisterTransitGatewayMulticastGroupSourcesCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeregisterTransitGatewayMulticastGroupSourcesCommand = void 0;
const models_2_1 = require("../models/models_2");
const Aws_ec2_1 = require("../protocols/Aws_ec2");
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
/**
 * <p>Deregisters the specified sources (network interfaces) from the  transit gateway multicast group.</p>
 */
class DeregisterTransitGatewayMulticastGroupSourcesCommand extends smithy_client_1.Command {
    // Start section: command_properties
    // End section: command_properties
    constructor(input) {
        // Start section: command_constructor
        super();
        this.input = input;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(middleware_serde_1.getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "EC2Client";
        const commandName = "DeregisterTransitGatewayMulticastGroupSourcesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_2_1.DeregisterTransitGatewayMulticastGroupSourcesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: models_2_1.DeregisterTransitGatewayMulticastGroupSourcesResult.filterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return Aws_ec2_1.serializeAws_ec2DeregisterTransitGatewayMulticastGroupSourcesCommand(input, context);
    }
    deserialize(output, context) {
        return Aws_ec2_1.deserializeAws_ec2DeregisterTransitGatewayMulticastGroupSourcesCommand(output, context);
    }
}
exports.DeregisterTransitGatewayMulticastGroupSourcesCommand = DeregisterTransitGatewayMulticastGroupSourcesCommand;
