// @aws-sdk/client-ec2/dist/cjs/commands/SearchTransitGatewayRoutesCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchTransitGatewayRoutesCommand = void 0;
const models_5_1 = require("../models/models_5");
const Aws_ec2_1 = require("../protocols/Aws_ec2");
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
/**
 * <p>Searches for routes in the specified transit gateway route table.</p>
 */
class SearchTransitGatewayRoutesCommand extends smithy_client_1.Command {
    // Start section: command_properties
    // End section: command_properties
    constructor(input) {
        // Start section: command_constructor
        super();
        this.input = input;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(middleware_serde_1.getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "EC2Client";
        const commandName = "SearchTransitGatewayRoutesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_5_1.SearchTransitGatewayRoutesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: models_5_1.SearchTransitGatewayRoutesResult.filterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return Aws_ec2_1.serializeAws_ec2SearchTransitGatewayRoutesCommand(input, context);
    }
    deserialize(output, context) {
        return Aws_ec2_1.deserializeAws_ec2SearchTransitGatewayRoutesCommand(output, context);
    }
}
exports.SearchTransitGatewayRoutesCommand = SearchTransitGatewayRoutesCommand;
